applets/notification_area/main.c

```diff
@@ -104,6 +109,32 @@ get_gtk_orientation_from_applet_orient (MatePanelAppletOrient orient)
   return GTK_ORIENTATION_HORIZONTAL;
 }
 
+static void
+gsettings_changed_min_icon_size (GSettings    *settings,
+                                 gchar        *key,
+                                 NaTrayApplet *applet)
+{
+  applet->priv->min_icon_size = g_settings_get_int (settings, key);
+
+  set_grid_display_mode (NA_GRID (applet->priv->grid), applet->priv->min_icon_size);
```

Comment:
Ah, now I see the use for this function, but then you probably should add it in this commit, because in previous commits this function call looks unneeded. Also function name seems wrong and does not match what it does... To me looks like this should be called `na_grid_set_min_icon_size`

Reply:
as above, if I knew how to get rid of size-allocate in main.c, that call would simply be to provide the min_icon_size to the grid and hence wouldn't appear as a dud "et_grid_display_mode (NA_GRID (applet->priv->grid), MIN_ICON_SIZE_DEFAULT);" in the previous as at that point there's no size to change.. That call there is to update the grid, without it there the grid didn't update..